components/style/properties/properties.mako.rs

```diff
@@ -1554,6 +1556,14 @@ impl ComputedValues for ServoComputedValues {
             }
         }
 
+        fn clone_for_text_node(&self) -> Self {
+            // Text nodes get a copy of the parent style. This ensures
+            // that during fragment construction any non-inherited
+            // CSS properties (such as vertical-align) are correctly
+            // set on the fragment(s).
+            self.clone()
```

Comment:
Might be worth fleshing this comment out in a bit more detail, just explaining what gw and pcwalton told us (i.e. this is an implementation decision by servo layout, which isn't central to the design but would nonetheless be a pain to change).